Tasks/ANT/anttask.ts

```diff
@@ -77,6 +77,31 @@ function processAntOutputLine(line) {
     }
 }
 
+function readJavaHomeFromRegistry(jdkVersion: string, arch: string): string {
+    let reg = tl.tool('reg');
+    reg.arg(['query', `HKLM\\SOFTWARE\\JavaSoft\\Java Development Kit\\${jdkVersion}`, "/v", "JavaHome"]);
+    if (arch.toLowerCase() === "x86") {
+        reg.arg("/reg:32");
+    } else {
+        reg.arg("/reg:64");
+    }
+
+    let result = reg.execSync();
+    let javaHome = null;
+
+    if (result && result.code === 0 && result.stdout) {
+        let regSzIdx = result.stdout.trim().indexOf("REG_SZ");
+        if (regSzIdx > -1) {
+            let output: string[] = result.stdout.split("REG_SZ");
+            if (output.length === 2) {
+                javaHome = output[1].trim(); // value is what comes after
+                tl.debug("Java_Home: " + javaHome);
+            }
+        } 
+    }
+
+    return javaHome;
```

Comment:
Error handling is centralized in the main logic.  In my mind, utility methods should not handle errors as it may be normal that this value isn't found -- we could check reg first and fall back to variables later.